### PR TITLE
New metric: `LogAUC`

### LINKED ISSUE #2304: LogAUC metric
## 🚀 Feature

Area Under ROC with log transformed FPR.

### Motivation

LogAUC is similar to AUROC and is a popular and effective metric to evaluate classification models on very imbalanced datasets. It's used in Drug Design/Discovery applications.

### Alternatives

The code is already implemented [here](https://github.com/mims-harvard/TDC/blob/15809f26ef11d034b56077f2a930ce43806c5783/tdc/evaluator.py#L66).

### Additional context

The metric was introduced in this [paper](https://pubmed.ncbi.nlm.nih.gov/20735049/).

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #2304

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2377.org.readthedocs.build/en/2377/

### CODE REVIEW

**Comment by Borda** [comment]
@SkafteNicki, what is missing here to make it land? :chipmunk:

**Comment by Borda** [comment]
just resolved conflicts, let's make it land... :flamingo:

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
seems one of the needed packages is missing for py3.12
ref:https://github.com/kuelumbus/rdkit-pypi/pull/111#issuecomment-2455212668

File: src/torchmetrics/classification/logauc.py [hunk 1]
```diff
@@ -0,0 +1,507 @@
...
+    Args:
+        num_classes: Integer specifying the number of classes
+        fpr_range: 2-element tuple with the lower and upper bound of the false positive rate range to compute the log
+            AUC score.
+        average:
+            Defines the reduction that is applied over classes. Should be one of the following:
+
+            - ``macro``: Calculate score for each class and average them
+            - ``weighted``: calculates score for each class and computes weighted average using their support
+            - ``"none"`` or ``None``: calculates score for each class and applies no reduction
```
Comment (Borda) [inline src/torchmetrics/classification/logauc.py:200]:
```suggestion
            - ``"macro"``: Calculate score for each class and average them
            - ``"weighted"``: calculates score for each class and computes weighted average using their support
            - ``"none"`` or ``None``: calculates score for each class and applies no reduction
```

File: src/torchmetrics/classification/logauc.py [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+
+    Args:
+        num_labels: Integer specifying the number of labels
+        fpr_range: 2-element tuple with the lower and upper bound of the false positive rate range to compute the log
+            AUC score.
+        average:
+            Defines the reduction that is applied over labels. Should be one of the following:
+
+            - ``macro``: Calculate score for each label and average them
+            - ``"none"`` or ``None``: calculates score for each label and applies no reduction
```
Comment (Borda) [inline src/torchmetrics/classification/logauc.py:349]:
```suggestion
            - ``"macro"``: Calculate the score for each label and average them
            - ``"none"`` or ``None``: calculates score for each label and applies no reduction
```

File: requirements/classification_test.txt [hunk 3]
```diff
@@ -5,3 +5,4 @@ pandas >1.4.0, <=2.2.3
 netcal >1.0.0, <1.4.0 # calibration_error
 numpy <2.2.0
 fairlearn # group_fairness
+PyTDC ==0.4.1 ; python_version <"3.12"  # locauc, temporal_dependency
```
Comment (Borda) [inline requirements/classification_test.txt:8]: Let's drop this condition :)

Comment (amva13) [reply requirements/classification_test.txt:8]: @Borda thanks for pointing to this. newer versions of PyTDC should not encounter this problem as the rdkit-pypi dependency was removed. looks like you resolved anyhow.

**Pull request merged**